### PR TITLE
parts: allow debug shell after the lifecycle

### PR DESCRIPTION
**Pull request by cmatsuoka**
Shell into the instance if `--debug` is used and a packing error happens
after the execution of the parts lifecycle.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [ ] Have you successfully run `pytest tests/unit`?

-----
CRAFT-1286

### CODE REVIEW

**Comment by cmatsuoka** [comment]
@hellsworth Could you verify if this addresses the `--debug` issues you had?

**Comment by hellsworth** [comment]
@cmatsuoka Well maybe? I had saved off my test case in a local branch of the app that exhibited the failure after the lifecycle. But now when I do a clean build with this snapcraft, my snap builds. I've done a clean build a couple of times, both successfully building the snap.

**Comment by hellsworth** [comment]
@sergiusens pointed out that there was a cmake issue fixed with respect to how libraries are found. My project absolutely uses cmake, so perhaps this fix got in the way.

File: snapcraft/parts/lifecycle.py [hunk 1]
```diff
@@ -449,7 +490,9 @@ def _run_in_provider(
         except subprocess.CalledProcessError as err:
             capture_logs_from_instance(instance)
             raise errors.SnapcraftError(
-                f"Failed to execute {command_name} in instance."
+                f"Failed to execute {command_name} in instance. "
+                "Run the same command again with --debug to shell into "
```
Comment (sergiusens) [inline snapcraft/parts/lifecycle.py:494]: this should use the extra params from CraftError which SnapcraftError inherits from

**Review by sergiusens — APPROVED** [review]

**Pull request merged**